src/Microsoft.AspNetCore.Rewrite/Internal/ModRewrite/FlagParser.cs

```diff
@@ -2,22 +2,68 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 
 namespace Microsoft.AspNetCore.Rewrite.Internal.ModRewrite
 {
     public class FlagParser
-    { 
-        public static Flags Parse(string flagString)
+    {
+        private IDictionary<string, FlagType> _ruleFlagLookup = new Dictionary<string, FlagType>(StringComparer.OrdinalIgnoreCase) {
```

Comment:
Can we make it static and readonly so as to avoid repeatedly allocate this dictionary?